app/models/account.rb

```diff
@@ -76,7 +76,7 @@ class Account < ApplicationRecord
   validates :username, format: { with: /\A#{USERNAME_RE}\z/i }, if: -> { !local? && will_save_change_to_username? }
 
   # Local user validations
-  validates :username, format: { with: /\A[a-z0-9_]+\z/i }, length: { maximum: 30 }, if: -> { local? && will_save_change_to_username? && actor_type != 'Application' }
+  validates :username, format: { with: /\A[a-z0-9_]+\z/i }, length: { maximum: 30 }, if: -> { local? && will_save_change_to_username? && !%w(Application Group).include?(actor_type) }
```

Comment:
Maybe better to define `application_or_group?` method instead of doing a raw comparison inside a lambda.

Also this local-only validation change may be too early... It won't help with backwards-compatiblity later (since it's local only anyway) and we don't know how groups will be created/used UX-wise yet. For example, if groups will need to be mentionable, then we would need to conform to the same username regex. If groups will instead have special UI for addressing, then username won't matter.

Reply:
This change does not seem to be necessary yet. Let's re-suggest along with other features. Revert it.